### ISSUE TITLE
*arrayContainer.iorArray() always results in conversion to bitmap containers

### ISSUE THREAD

**Issue by maciej** [RoaringBitmap/roaring#106]
When computing a union of two array containers both array containers are converted to bitmap containers first, see [arraycontainer.go:342](https://github.com/RoaringBitmap/roaring/blob/master/arraycontainer.go#L342).
Intuitively it seems to be sub-optimal. In the Java roaring implementation a different approach is used ([see code](https://github.com/RoaringBitmap/RoaringBitmap/blob/master/roaringbitmap/src/main/java/org/roaringbitmap/ArrayContainer.java#L652)), one that takes into account the sum of cardinalities as heuristic to determine the type of the result container.

I believe it's a piece of code that could be performance optimised.

**Comment by lemire** [comment]
Yes, it is likely true that this particular code can be improved. Of course, one needs to benchmark the result.

Want to take a stab at it?

**Comment by maciej** [comment]
Yes, I'll look into that whenever I have the time.
